dataprofiler/data_readers/csv_data.py

```diff
@@ -103,24 +103,150 @@ def _check_and_return_options(self, options):
                                  'integer for the row that represents the '
                                  'header (0 based index)')
         return options
-
+    
     @staticmethod
-    def _determine_delimiter_of_str(data_as_str):
+    def _guess_delimiter(data_as_str, preferred=[',', '\t'], omitted=['"', "'"]):
         """
         Automatically checks for what delimiter exists in a text document.
 
         :param data_as_str: Single string containing rows (lines separated by "\n")
         :type data_as_str: str
+        :param preferred: Delimiters that will be weighted and selected if possible
+        :type preferred: list
+        :param omitted: Delimiters that will be omitted from selection
+        :type omitted: list
         :return: Delimiter, if none can be found None is returned
         :rtype: str or None
         """
-        sniffer = csv.Sniffer()
-        sniffer.preferred = [',', '\t', ';']  # removes ' ', ':' from preferred
-        try:
-            dialect = sniffer.sniff(data_as_str)
-        except csv.Error as exc:
-            return None
-        return dialect.delimiter
+
+        # Detect vocabulary (and count)
+        vocab = {}
+        for i in range(len(data_as_str)):
+            c = data_as_str[i]
+            if c not in vocab:
+                vocab[c] = 0
+            vocab[c] += 1            
```

Comment:
python has a builtin counter in collections. Might be better to use or potentially optimized somehow?